### DISCUSSION TITLE
Setting up TCP Socket on Shared Hosting

### DISCUSSION THREAD

**Discussion by MMahyaRA** [reactphp/reactphp#560] [Q&A]
Hello, I want to establish a TCP socket using this library on shared hosting. My issue is that I don't have access to the console to install Composer, and I cannot install it that way. How can I use this library? Is there a way similar to the C language where I can place the prerequisite files in the main file path and then call them with the main file?

**Comment by SimonFrings** [comment]
Hey @MMahyaRA, the easiest way for installing this project is definitely through Composer, so if your hosting service can't provide you access to the command line, my first suggestion would be to use a different hoster. Not sure what you're currently using, but I know that hosting services like netcup and others work well.

However, if it isn't an option to change your hosting service, you can install Composer on your local machine and then use it to install all necessary dependencies for your project. For this, you need a `composer.json` file, see https://getcomposer.org/doc/01-basic-usage.md for more details. Installing the dependencies will create a folder named `vendor` which will contain all the necessary project dependency files and will also create a `composer.lock` file.

You can then copy the `composer.json` file, the `composer.lock` file and the `vendor` directory to your hoster and you should be able to use the installed libraries. Note however, that you should always make sure that the requirements on your local machine (PHP version, extensions, etc.) match your production server requirements.

I hope this helps :+1:
